Library/Formula/ocilib.rb

```diff
@@ -0,0 +1,41 @@
+class OracleHomeVar < Requirement
+  fatal true
+  satisfy ENV["ORACLE_HOME"]
+
+  def message; <<-EOS.undent
+      To use this formula you have to set the ORACLE_HOME environment variable.
+      Check Oracle Instant Client documentation for more information.
+    EOS
+  end
+end
+
+class Ocilib < Formula
+  desc "Open source C and C++ library for accessing Oracle Databases"
+  homepage "http://www.ocilib.net/"
+  url "https://downloads.sourceforge.net/project/orclib/OCILIB%20Sources/4.1.0/ocilib-4.1.0-gnu.tar.gz"
+  mirror "https://www.mirrorservice.org/sites/ftp.sourceforge.net/pub/sourceforge/o/or/orclib/OCILIB%20Sources/4.1.0/ocilib-4.1.0-gnu.tar.gz"
+  sha256 "d86c1e6d5ef33541299996b76497a1271498cdcf5720ee0f1248a0b78dfb3a0c"
+
+  head "http://git.code.sf.net/p/orclib/git"
+
+  depends_on OracleHomeVar
+
+  def install
+    system "./configure"
+    system "make"
+    system "make", "install"
```

Comment:
Thanks @xu-cheng, I've added it along with some more required (and optional) settings.